modules/configuration/src/main/java/org/apache/ignite/internal/configuration/asm/ConfigurationAsmGenerator.java

```diff
@@ -487,7 +488,20 @@ static String fieldName(Field f) {
     public static String publicName(Field f) {
         PublicName annotation = f.getAnnotation(PublicName.class);
 
-        return annotation == null ? f.getName() : annotation.value();
+        return annotation == null || "".equals(annotation.value()) ? f.getName() : annotation.value();
```

Comment:
Could have used `isEmpty()` instead, usually we use `"...".equals(...)` if the second part is nullable. It's not the case here